packages/core/streams/streaming-text-response.ts

```diff
@@ -16,29 +16,116 @@ export class StreamingTextResponse extends Response {
   }
 }
 
+/**
+ * A simple utility function to convert a ReadableStream to a Node.js Readable.
+ * This is campatible with Node.js 18 web streams and polyfill-stream package both.
+ */
+export function streamToNodeReadable(stream: ReadableStream) {
+  const reader = stream.getReader()
+  let closed = false
+
+  // in order to avoid module check error
+  const Readable = require('stream')
+    .Readable as typeof import('stream').Readable
+
+  const readable = new Readable({
+    read() {
+      reader
+        .read()
+        .then(({ done, value }) => {
+          if (done) {
+            this.push(null)
+          } else {
+            this.push(value)
+          }
+        })
+        .catch(e => {
+          readable.destroy(e)
+        })
```

Comment:
Let's use async-await to save a few promise allocations:
```suggestion
    async read() {
      try {
        const { done, value } = await reader.read()
        if (done) {
          this.push(null)
        } else {
          this.push(value)
        }
      } catch (e) {
        readable.destroy(e as Error)
      }
```